google-cloud-logging/lib/google/cloud/logging/middleware.rb

```diff
@@ -52,23 +54,26 @@ class Middleware
         # @return [Google::Cloud::Logging::Middleware] A new
         #   Google::Cloud::Logging::Middleware instance
         #
-        def initialize app, logger: nil, **kwargs
+        def initialize app, logger: nil, on_init: nil, **kwargs
           @app = app
 
           load_config kwargs
 
-          if logger
-            @logger = logger
-          else
+          logger ||= Middleware.logger
+          logger ||= begin
             log_name = configuration.log_name
             logging = Logging.new project_id: configuration.project_id,
                                   credentials: configuration.credentials
             resource = Middleware.build_monitored_resource(
               configuration.monitored_resource.type,
               configuration.monitored_resource.labels
             )
-            @logger = logging.logger log_name, resource
+            Middleware.logger = logging.logger log_name, resource
           end
+
+          on_init.call if on_init
```

Comment:
Thoughts on this callback not being passed any arguments?

Reply:
I think it's fine. The logger can be obtained from the Middleware class. If we do decide later that we need to pass something to it, users' existing no-argument procs will still work.